lib/fs.js

```diff
@@ -293,7 +303,12 @@ ReadFileContext.prototype.close = function(err) {
   req.oncomplete = readFileAfterClose;
   req.context = this;
   this.err = err;
-  binding.close(this.fd, req);
+
+  if (this.isUserFd) {
+    req.oncomplete.call(req, null);
```

Comment:
Correct. No `fd` ownership transfer takes place. We did not create the `fd`, so we should not close it IMHO. Doing so would prevent users from re-using existing file descriptors more than once, pretty much defeating the point of this PR.